fs-repo-11-to-12/vendor/github.com/marten-seemann/qtls-go1-15/common.go

```diff
@@ -430,6 +431,10 @@ type clientHelloInfo struct {
 	// config is embedded by the GetCertificate or GetConfigForClient caller,
 	// for use with SupportsCertificate.
 	config *Config
+
+	// NOTE this field was added in go 1.18
```

Comment:
or go 1.17?

Reply:
source
https://github.com/golang/go/blob/6ad27161f8d1b9c5e03fb3415977e1d3c3b11323/src/crypto/tls/common.go#L450

blame
https://github.com/golang/go/commit/860704317e02d699e4e4a24103853c4782d746c1

so this was released in go 1.17